scripts/extensions/auto-tagging-widget/src/header-auto-tagging.tsx

```diff
@@ -0,0 +1,90 @@
+    const groupLabels = getGroups(superdesk);
+
+    return class HeaderAutoTagging extends React.PureComponent<IProps> {
+        private iMatricsFields = superdesk.instance.config.iMatricsFields ?? {entities: {}, others: {}};
+
+        render() {
+            const existingTags = getExistingTags(this.props.article);
+            const resClient = toClientFormat(existingTags);
+            const data = {original: {analysis: resClient}, changes: {analysis: resClient}};
+
+            const {entitiesGroupedAndSorted, othersGrouped} = getAutoTaggingData(data, this.iMatricsFields);
+
+            const savedTags = data.original.analysis.keySeq().toSet();
+
+            let allGrouped = OrderedMap<string, JSX.Element>();
+
+            othersGrouped.forEach((tags, groupId) => {
+                if (tags != null && groupId != null) {
+                    allGrouped = allGrouped.set(groupId,
+                        <div>
+                            <TagListComponent
+                                savedTags={savedTags}
+                                tags={tags.toMap()}
+                                readOnly={true}
+                                inline={true}
+                                onRemove={() => true}
+                            />
+                        </div>,
+                    );
+                }
+            });
+
+            if (entitiesGroupedAndSorted.size > 0) {
+                allGrouped = allGrouped.set('entities',
+                    <div>
+                        {entitiesGroupedAndSorted.map((tags, key) => (
+                            <div key={key}>
+                                <div
+                                    className="form-label"
+                                    style={{display: 'block'}}
+                                >
+                                    {groupLabels.get(key).plural}
+                                </div>
+                                <TagListComponent
+                                    savedTags={savedTags}
+                                    tags={tags.toMap()}
+                                    readOnly={true}
+                                    inline={true}
+                                    onRemove={() => true}
+                                />
+                            </div>
+                        )).toArray()}
+                    </div>,
+                );
+            }
+
+            const allGroupedAndSortedByConfig = allGrouped
+                .filter((_, key) => hasConfig(key, this.iMatricsFields.others))
+                .sortBy((_, key) => this.iMatricsFields.others[key].order,
+                    (a, b) => a - b);
+
+            const allGroupedAndSortedNotInConfig = allGrouped
+                .filter((_, key) => !hasConfig(key, this.iMatricsFields.others));
+
+            const allGroupedAndSorted = allGroupedAndSortedByConfig
+                .concat(allGroupedAndSortedNotInConfig);
```

Comment:
Can you extract this code into a reusable piece so it's not duplicated in multiple places? I would also consider not putting JSX into a map, but grouping raw objects and outputting JSX only in the return block of the `render` method.